src/mem/alloc.h

```diff
@@ -179,26 +186,27 @@ namespace snmalloc
       size_t size_bits = bits::next_pow2_bits(size);
       set(p, (uint8_t)size_bits);
       // Set redirect slide
-      uintptr_t ss = (uintptr_t)((size_t)p + SUPERSLAB_SIZE);
+      uintptr_t ss = (uintptr_t)p + SUPERSLAB_SIZE;
       for (size_t i = 0; i < size_bits - SUPERSLAB_BITS; i++)
       {
         size_t run = 1ULL << i;
         PagemapProvider::pagemap().set_range(
-          (void*)ss, (uint8_t)(64 + i + SUPERSLAB_BITS), run);
-        ss = (uintptr_t)ss + SUPERSLAB_SIZE * run;
+          ss, (uint8_t)(64 + i + SUPERSLAB_BITS), run);
+        ss = ss + SUPERSLAB_SIZE * run;
       }
-      PagemapProvider::pagemap().set(p, (uint8_t)size_bits);
+      PagemapProvider::pagemap().set((uintptr_t)p, (uint8_t)size_bits);
     }
     /**
      * Update the pagemap to remove a large allocation, of `size` bytes from
      * address `p`.
      */
-    void clear_large_size(void* p, size_t size)
+    void clear_large_size(void* vp, size_t size)
     {
+      uintptr_t p = (uintptr_t)vp;
```

Comment:
`reinterpret_cast`.

We should probably add clang-tidy to our CI...

Reply:
That would be good.